optimized_ingestion/stages/detection_estimation/__init__.py

```diff
@@ -68,15 +77,19 @@ def _run(self, payload: "Payload"):
         assert dets is not None, [*payload.metadata.keys()]
         metadata: "list[DetectionEstimationMetadatum]" = []
         start_time = time.time()
+        investigation_frame_nums = []
         for i in Stage.tqdm(range(len(payload.video) - 1)):
             current_ego_config = payload.video[i]
+            current_fps = math.ceil(
+                1 / (payload.video[i + 1].timestamp - current_ego_config.timestamp).total_seconds())
             if i != next_frame_num:
```

Comment:
I'm utilizing the fps to bridge the gap between time and frame. Not ideal but it works